float32 posX
float32 posY
float32 posZ
time timestamp
